common/autoware_lanelet2_ros_interface/lib/query.cpp

```diff
@@ -32,6 +32,42 @@ namespace lanelet
 namespace utils
 {
 
+void overwriteWithMatchingId (WeakLanelet& prim, const lanelet::LaneletMapPtr ll_Map)
+{
+  if (ll_Map->laneletLayer.exists(prim.lock().id()))
+  {
+    prim = ll_Map->laneletLayer.get(prim.lock().id());
```

Comment:
Generally when dealing with things that 'lock' you want to acquire the lock perform all your operations then release the lock. While this isn't currently being run in a multi-threaded context if it was this "if" check would still be susceptible to race conditions. 

Something like this is preferable
```
if (auto prim_ptr = prim.lock() && ll_Map->laneletLayer.exists(prim_ptr->id()) {
    prim = ll_Map->laneletLayer.get(prim_ptr->id());
} else {
 // log could not acquire weak pointer lock or primitive did not exist in the map
}

Reply:
Added